### ISSUE TITLE
fake_data与ori_data差距较大的问题

### ISSUE THREAD

**Issue by kilig000123** [Y-debug-sys/Diffusion-TS#103]
感谢作者的分享和回答！
我在自己的数据集上尝试生成fake_data与原数据比较时发现差距较大

![Image](https://github.com/user-attachments/assets/92e65eed-d2a9-4d24-a444-83c96af2bc7a)

![Image](https://github.com/user-attachments/assets/74883a56-f0ed-4673-837e-671802aaf9e6)

![Image](https://github.com/user-attachments/assets/d1616304-dad4-4f21-b70a-a4652ef3862a)

我使用的是一个类似solar的数据集，但是每天采样了144个，即周期144，我沿用了你在Utils\Data_utils\real_datasets.py中的类，config文件如下：

`model:
  target: Models.interpretable_diffusion.gaussian_diffusion.Diffusion_TS
  params:
    seq_length: 144
    feature_size: 128
    n_layer_enc: 4
    n_layer_dec: 4
    d_model: 96
    timesteps: 500
    sampling_timesteps: 500
    loss_type: 'l1'
    beta_schedule: 'cosine'
    n_heads: 4
    mlp_hidden_times: 4
    attn_pd: 0.0
    resid_pd: 0.0
    kernel_size: 1
    padding_size: 0

solver:
  base_lr: 1.0e-5
  max_epochs: 1500
  results_folder: ./Checkpoints_solor_pic
  gradient_accumulate_every: 2
  save_cycle: 150  # max_epochs // 10
  ema:
    decay: 0.995
    update_interval: 10
  
  scheduler:
    target: engine.lr_sch.ReduceLROnPlateauWithWarmup
    params:
      factor: 0.5
      patience: 300
      min_lr: 1.0e-5
      threshold: 1.0e-1
      threshold_mode: rel
      warmup_lr: 8.0e-4
      warmup: 100 
      verbose: False

dataloader:
  train_dataset:
    target: Utils.Data_utils.real_datasets.CustomDataset
    params:
      name: solar_pic
      proportion: 0.9  # Set to rate < 1 if training conditional generation
      data_root: ./Data/solar_test.csv
      window: 144  # seq_length
      save2npy: True
      neg_one_to_one: True
      seed: 1234
      period: train

  test_dataset:
    target: Utils.Data_utils.real_datasets.CustomDataset
    params:
      name: solar_pic
      proportion: 0.9  # rate
      data_root: ./Data/solar_test.csv
      window: 144  # seq_length
      save2npy: True
      neg_one_to_one: True
      seed: 1234
      period: test
      style: separate
      distribution: geometric
    coefficient: 1.0e-2
    step_size: 5.0e-2
    sampling_steps: 200

  batch_size: 128
  sample_size: 128
  shuffle: True`

可以帮我看一下问题出现在什么地方吗，希望得到你的帮助

**Comment by Y-debug-sys** [comment]
你好，训练时间太短了

**Comment by kilig000123** [comment]
1500个epoch后loss为0.066382，请问你觉得可以增加到多大比较合适呢？

**Comment by Y-debug-sys** [comment]
这不是epoch，是反向传播次数，把这个值放大到和你的数据集长度成正比（此外，其它关于训练的周期参数也要适当放大），可以参考其它的config。

**Comment by kilig000123** [comment]
感谢你的回复，timesteps 和 sampling_timesteps 这两个扩散步长的参数需要增大吗

**Comment by Y-debug-sys** [comment]
不太需要
